specification/appconfiguration/AppConfiguration/models.tsp

```diff
@@ -151,6 +151,36 @@ model Error {
   status?: int32;
 }
 
+@doc("An object containing specific information about an error.")
+@error
+model InnerError {
+  @doc("One of a server-defined set of error codes.")
+  code?: string;
+
+  @doc("An object containing more specific information than the current object about the error.")
+  #suppress "@azure-tools/typespec-azure-core/property-name-conflict" "Property 'innererror' having the same name as its enclosing model will cause problems with C# code generation."
+  innererror?: InnerError;
+}
+
+@doc("The details of an error.")
+@error
+model ErrorDetail {
```

Comment:
why cant we just do `model ErrorDetail is Foundations.Error` ?

Reply:
Got this error when compile: 
![image](https://github.com/user-attachments/assets/e0b01e8e-3da9-4d57-bea1-e38a61e86a3a)

Reply:
These errors look like you tried to define the model inline on the property which is not supported syntax.

Reply:
you just need to define the model independently and then refer to it in the property field

Reply:
If using `model ErrorDetails is Foundations.Error` is too complicated to get to work with the emitters, then let's make sure the models you add to the spec match the service's rest api.

Reply:
The models are consisting with what we are using now. We can use `Foundations.Error` when emitter supports renaming on it.